pyomo/contrib/parmest/tests/test_new_parmest_capabilities.py

```diff
@@ -0,0 +1,282 @@
+
+            m.hour = pyo.Var(within=pyo.PositiveReals, initialize=0.1)
+            m.y = pyo.Var(within=pyo.NonNegativeReals)
+
+            @m.Constraint()
+            def response_rule(m):
+                return m.y == m.asymptote * (1 - pyo.exp(-m.rate_constant * m.hour))
+
+            return m
+
+        # create the Experiment class
+        class RooneyBieglerExperiment(Experiment):
+            def __init__(self, experiment_number, hour, y):
+                self.y = y
+                self.hour = hour
+                self.experiment_number = experiment_number
+                self.model = None
+
+            def get_labeled_model(self):
+                if self.model is None:
+                    self.create_model()
+                    self.finalize_model()
+                    self.label_model()
+                return self.model
+
+            def create_model(self):
+                m = self.model = rooney_biegler_model()
+
+                return m
+
+            def finalize_model(self):
+                m = self.model
+
+                # fix the input variable
+                m.hour.fix(self.hour)
+
+                return m
+
+            def label_model(self):
+                m = self.model
+
+                # add experiment inputs
+                m.experiment_inputs = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.experiment_inputs.update([(m.hour, self.hour)])
+
+                # add experiment outputs
+                m.experiment_outputs = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.experiment_outputs.update([(m.y, self.y)])
+
+                # add unknown parameters
+                m.unknown_parameters = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.unknown_parameters.update(
+                    (k, pyo.value(k)) for k in [m.asymptote, m.rate_constant]
+                )
+
+                # add measurement error
+                m.measurement_error = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+                m.measurement_error.update([(m.y, None)])
+
+                return m
+
+        # creat the experiments list
+        rooney_biegler_exp_list = []
+        hour_data = self.data["hour"]
+        y_data = self.data["y"]
+        for i in range(self.data.shape[0]):
+            rooney_biegler_exp_list.append(
+                RooneyBieglerExperiment(i, hour_data[i], y_data[i])
+            )
+
+        self.exp_list = rooney_biegler_exp_list
+
+        self.objective_function = (
+            "SSE"  # testing the new covariance calculations for the `SSE` objective
+        )
+
+    def check_rooney_biegler_results(self, objval, cov):
+        """
+        Checks if the results are equal to the expected values and agree with the results of Rooney-Biegler
+
+        Argument:
+            objval: the objective value of the annotated Pyomo model
+            cov: covariance matrix of the estimated parameters
```

Comment:
I would recommend `obj_val` rather than `objval` - it's more accessible on screen readers.